scripts/jenkins/cloud/manual/Dockerfile

```diff
@@ -0,0 +1,53 @@
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE/Updates/SLE-Live-Patching/12/x86_64/update/                   live-patching
+RUN zypper addrepo --refresh http://clouddata.cloud.suse.de/repos/x86_64/SLES12-SP4-Pool/                                   pool
+RUN zypper addrepo --refresh http://clouddata.cloud.suse.de/repos/x86_64/SLES12-SP4-Updates/                                up
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12-SP4:/GA/standard/                                     12-SP4-GA
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12-SP4:/Update/standard/                                 12-SP4-Update
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12-SP3:/GA/standard/                                     12-SP3-GA
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12-SP3:/Update/standard/                                 12-SP3-Update
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/GA/standard/                                     12-SP2-GA
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update/standard/                                 12-SP2-Update
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12-SP1:/GA/standard/                                     12-SP1-GA
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12-SP1:/Update/standard/                                 12-SP1-Update
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12:/GA/standard/                                         12-GA
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/SLE-12:/Update/standard/                                     12-Update
+RUN zypper addrepo --refresh http://download.suse.de/ibs/SUSE:/Factory:/Head/standard/                                      factory
+
+RUN zypper --gpg-auto-import-keys refresh
+
+RUN zypper --non-interactive --gpg-auto-import-keys install \
+        autoconf \
+        automake \
+        ca-certificates-suse \
+        gcc \
+        git-core \
+        python-devel \
+        python-virtualenv \
+        python3 \
+        python3-devel \
+        python3-virtualenv \
+        sshpass
+
+ARG python
+RUN virtualenv --python ${python:-python2.7} /opt/ansible-venv
+RUN /opt/ansible-venv/bin/pip install --upgrade pip
+COPY requirements.txt /tmp/requirements.txt
+RUN /opt/ansible-venv/bin/pip install --requirement /tmp/requirements.txt
+RUN /opt/ansible-venv/bin/pip install python-openstackclient python-heatclient
+
+ARG git_user_name
+ARG git_user_email
+RUN git config --global user.name "${git_user_name:-Unknown User}"
+RUN git config --global user.email "${git_user_email:-unknown@unknown.org}"
+
+VOLUME ["/opt/automation"]
+WORKDIR /opt/automation/scripts/jenkins/cloud/manual
+ENV ANSIBLE_VENV /opt/ansible-venv
+ENV CLIFF_FIT_WIDTH=1
+RUN ["/bin/bash", "-c", "echo . /opt/ansible-venv/bin/activate >> /root/.bashrc"]
```

Comment:
Does this actually take effect when you run the container after building it? I think you'll still have to activate the venv at runtime.

Reply:
It's active when I run the container. But now that you mention it, it's kind of strange that it does.

Reply:
Even if it does somehow persist, I would not use the Dockerfile to activate the venv. I would defer that to runtime by letting the entrypoint activate it.

Reply:
This works because it uses bash to activate the venv. I moved the activation into the entrypoint.